WikiWatershed/model-my-watershed
# Upgrade Front-end Dependencies

## Overview

Upgrades a number of dependencies to resolve security issues. Nunjucks has not been upgraded and is relegated to #3140.

Connects #3182 

## Testing Instructions

* Check out this branch
* Remove your `node_modules` `$ rm -rf src/mmw/node_modules`
* Destroy and recreate `app` VM `$ vagrant destroy -f app && vagrant up`
* Rebundle everything `$ ./scripts/bundle.sh --debug --vendor --tests`
* Ensure tests run successfully `$ ./scripts/test.sh`
* Ensure project still works correctly by clicking around
* Ensure bundle watching still works by running `$ ./scripts/bundle.sh --watch` and making changes to the codebase (maybe change the title [here](https://github.com/WikiWatershed/model-my-watershed/blob/develop/src/mmw/js/src/draw/settings.js#L6)) and saving the file, and reloading the page to see your changes
* Ensure linting still works `$ vagrant ssh app -c 'cd /opt/app && yarn run lint'`
* Do a security audit and ensure there's no issues outside of Nunjucks `$ vagrant ssh app -c 'cd /opt/app && yarn audit'`
